Podverse/PVAuth.swift

```diff
@@ -7,222 +7,104 @@
-////                    clip.endTime = endTime
-////                }
-////                
-////                if let dateCreated = dictResponse["dateCreated"] as? String {
-////                    clip.dateCreated = PVUtility.formatStringToDate(dateCreated)
-////                }
-////                
-////                if let lastUpdated = dictResponse["lastUpdated"] as? String {
-////                    clip.lastUpdated = PVUtility.formatStringToDate(lastUpdated)
-////                }
-////                
-////                if let serverEpisodeId = dictResponse["episodeId"] as? NSNumber {
-////                    clip.serverEpisodeId = serverEpisodeId
-////                }
-////                
-////                CoreDataHelper.saveCoreData(moc, completionBlock: { (saved) in
-////                    dispatch_group_leave(dispatchGroup)
-////                })
-////            }) { (error) -> Void in
-////                print("Not saved to server. Error: ", error?.localizedDescription)
-////                CoreDataHelper.saveCoreData(moc, completionBlock: nil)
-////                }.call()
-//            
-//        }
-//        
-//        if ownedPlaylistsArray.count < 1 && ownedClipsArray.count < 1 {
-//            dispatchGroup.enter()
-//            dispatchGroup.leave()
-//        }
-        
-        dispatchGroup.notify(queue: DispatchQueue.main) {
-            UserDefaults.standard.set(idToken, forKey: "idToken")
-            UserDefaults.standard.set(userId, forKey: "userId")
-//            TODO:
-//            PlaylistManager.shared.getMyPlaylistsFromServer({
-//                PlaylistManager.shared.createDefaultPlaylists()
-//            })
-//            
-            self.delegate?.authFinished()
-            if let cBlock = completionBlock {
-                cBlock()
-            }
+    func syncUserInfoWithServer () {
+        if let idToken = UserDefaults.standard.string(forKey: "idToken"), let userId = UserDefaults.standard.string(forKey: "userId") {
+            Auth0
+                .users(token: idToken)
+                .get(userId, fields: ["nickname"], include: true)
+                .start { result in
+                    switch result {
+                    case .success(let user):
+                        guard let nickname = user["nickname"] as? String else {
+                            return
+                        }
+                        
+                        self.setUserInfo(idToken: idToken, userId: userId, userName: nickname)
+                    case .failure(let error):
+                        print(error)
+                    }
+                }
+            
         }
     }
     
-    func setUserNameAndUpdateOwnedItems(userName: String?) {
-        guard let idToken = UserDefaults.standard.string(forKey: "idToken") else {
-            return
-        }
+    func setUserInfo(idToken: String, userId: String?, userName: String?) {
```

Comment:
Please call this method something else other than "set..." Calling a method "set..." can overlap with setters (same as "get..."). Something like `populateUserInfoWith()` is my suggestion.